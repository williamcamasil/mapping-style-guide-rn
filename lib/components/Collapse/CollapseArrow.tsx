import React from 'react';

import { Icons } from '../../assets';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';

type CollapseArrowPropsType = {
  theme: AppThemeType;
  expanded?: boolean;
};

const CollapseArrow: React.FC<CollapseArrowPropsType> = ({
  theme, expanded,
}) => {
  const Icon = expanded ? Icons.Small.Top : Icons.Small.Down;
  return (
    <Icon width={16} height={16} color={theme.colors.neutralGray400} />
  );
};

export default withTheme(CollapseArrow);
