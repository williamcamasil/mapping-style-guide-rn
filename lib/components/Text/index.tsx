import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
