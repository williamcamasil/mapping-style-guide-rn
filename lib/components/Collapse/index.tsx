import React, {
  useCallback,
  useRef,
  useState,
} from 'react';
import {
  Animated,
  ColorValue,
  StyleProp,
  TouchableOpacity,
  LayoutChangeEvent,
  View,
  ViewStyle,
} from 'react-native';

import { useDidMountAndUpdate } from 'mapping-context-rn';

import { useTextStyles, useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Text from '../Text';
import CollapseArrow from './CollapseArrow';

export type CollapsePropsType = {
  theme: AppThemeType;
  title: string;
  children?: React.ReactNode;
  testID?: string;
  expanded?: boolean;
  description?: string;
  titleNumberOfLines?: number;
  style?: StyleProp<ViewStyle>;
  borderColor?: ColorValue;
  onPress?: () => void;
  onCollapseEnd?: () => void;
};

const Collapse: React.FC<CollapsePropsType> = ({
  children,
  theme,
  testID,
  expanded,
  title,
  description,
  titleNumberOfLines,
  style,
  borderColor,
  onPress,
  onCollapseEnd,
  ...others
}) => {
  const [collapseExpanded, setCollapseExpanded] = useState(Boolean(expanded));
  const animationHeight = useRef(new Animated.Value(0));
  const animationOpacity = useRef(new Animated.Value(0));
  const [headerHeight, setHeaderHeight] = useState(0);
  const [contentHeight, setContentHeight] = useState(0);

  useDidMountAndUpdate(() => {
    setCollapseExpanded(Boolean(expanded));
  }, [expanded]);

  const handleContentLayout = useCallback((event: LayoutChangeEvent) => {
    setContentHeight(event.nativeEvent.layout.height);
  }, []);

  const handleHeaderLayout = useCallback((event: LayoutChangeEvent) => {
    const height = event.nativeEvent.layout.height + theme.spacings.sMedium;
    setHeaderHeight(height);
    animationHeight.current.setValue(height);
  }, [theme.spacings.sMedium]);

  useDidMountAndUpdate(() => {
    const newHeight = collapseExpanded ? contentHeight : headerHeight;

    if (!newHeight) return undefined;

    const newOpacity = collapseExpanded ? 1 : 0;

    const animation = Animated.parallel([
      Animated.spring(animationHeight.current, {
        toValue: newHeight,
        useNativeDriver: false,
      }),
      Animated.spring(animationOpacity.current, {
        toValue: newOpacity,
        useNativeDriver: true,
      }),
    ]);

    animation.start(() => {
      onCollapseEnd?.();
    });

    return () => {
      animation.stop();
    };
  }, [collapseExpanded, contentHeight, headerHeight]);

  const handleCollapse = useCallback(() => {
    onPress?.();
    setCollapseExpanded(current => !current);
  }, [onPress]);

  const contentStyles = useViewStyles(() => [
    {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
    },
  ], []);

  const containerStyles = useViewStyles(() => [
    {
      overflow: 'hidden',
      height: contentHeight ? animationHeight.current : undefined,
      borderRadius: theme.borders.radius.medium,
      borderWidth: theme.borders.width.thin,
      borderColor: borderColor || theme.colors.neutralGray200,
    },
    style,
  ], [
    borderColor,
    contentHeight,
    style,
    theme.borders.radius.medium,
    theme.borders.width.thin,
    theme.colors.neutralGray200,
  ]);

  const headerTouchableStyles = useViewStyles(() => [
    {
      padding: theme.spacings.sMedium,
      paddingBottom: 0,
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
  ], [theme.spacings.sMedium]);

  const descriptionContainerStyles = useViewStyles(() => [
    {
      paddingHorizontal: theme.spacings.sMedium,
      marginTop: theme.spacings.sNano,
      opacity: animationOpacity.current,
    },
  ], [theme.spacings.sMedium, theme.spacings.sNano]);

  const titleStyles = useTextStyles(() => [
    {
      marginRight: theme.spacings.sQuark,
    },
  ], [theme.spacings.sQuark]);

  const childrenContainerStyles = useViewStyles(() => [
    {
      opacity: animationOpacity.current,
      padding: theme.spacings.sMedium,
      paddingTop: 0,
    },
  ], [theme.spacings.sMedium]);

  const renderDescription = () => {
    if (!description) {
      return null;
    }
    return <Text testID="collapse-description" variant="body" color="neutralGray600">{description}</Text>;
  };

  return (
    <Animated.View testID={testID} style={containerStyles}>
      <View testID="collapse-content" style={contentStyles} onLayout={handleContentLayout}>
        <TouchableOpacity
          {...others}
          testID="collapse-touchable"
          activeOpacity={theme.opacities.intense}
          onLayout={handleHeaderLayout}
          onPress={handleCollapse}
          style={headerTouchableStyles}
        >
          <Text
            testID="collapse-title"
            weight="bold"
            style={titleStyles}
            variant="body"
            color={collapseExpanded ? 'neutralGray700' : 'neutralGray600'}
            numberOfLines={titleNumberOfLines}
          >
            {title}
          </Text>
          <CollapseArrow expanded={collapseExpanded} />
        </TouchableOpacity>
        <Animated.View style={descriptionContainerStyles} testID="collapse-description-container">
          {renderDescription()}
        </Animated.View>
        <Animated.View testID="collapse-children-container" style={childrenContainerStyles}>
          {children}
        </Animated.View>
      </View>
    </Animated.View>
  );
};

Collapse.defaultProps = {
  titleNumberOfLines: 1,
};

export default withTheme(Collapse);
