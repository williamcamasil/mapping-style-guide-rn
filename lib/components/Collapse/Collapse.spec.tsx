import React from 'react';

import {
  fireEvent, render, waitFor,
} from '@testing-library/react-native';

import Collapse, { CollapsePropsType } from '.';
import { DefaultTheme } from '../../theme';
import Text from '../Text';

const fakeHeaderLayoutEvent = {
  nativeEvent: {
    layout: {
      height: 50,
    },
  },
};

const fakeContentLayoutEvent = {
  nativeEvent: {
    layout: {
      height: 200,
    },
  },
};

const renderCollapse = async (props: Omit<CollapsePropsType, 'theme'>) => {
  const handleCollapseEnd = jest.fn();

  const result = render(
    <Collapse
      {...props}
      onCollapseEnd={() => {
        handleCollapseEnd();
        props.onCollapseEnd?.();
      }}
    />,
  );

  const collapseContent = result.getByTestId('collapse-content');
  const collapseTouchable = result.getByTestId('collapse-touchable');

  fireEvent(collapseTouchable, 'layout', fakeHeaderLayoutEvent);
  fireEvent(collapseContent, 'layout', fakeContentLayoutEvent);

  await waitFor(() => {
    expect(handleCollapseEnd).toBeCalled();
  }, { timeout: 2000 });

  return result;
};

describe('Collapse snapshots', () => {
  it('default', async () => {
    const result = await renderCollapse({
      title: 'My title',
    });
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('description', async () => {
    const result = await renderCollapse({
      title: 'My title',
      description: 'My description',
    });
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('children', async () => {
    const result = await renderCollapse({
      title: 'My title',
      description: 'My description',
      children: (
        <Text>My children</Text>
      ),
    });
    expect(result.toJSON()).toMatchSnapshot();
  });

  it('expanded', async () => {
    const result = await renderCollapse({
      title: 'My title',
      description: 'My description',
      children: (
        <Text>My children</Text>
      ),
      expanded: true,
    });

    expect(result.toJSON()).toMatchSnapshot();
  });
});

describe('Collapse events', () => {
  it('collapsed height', async () => {
    const result = await renderCollapse({
      testID: 'collapse',
      title: 'My title',
      description: 'My description',
      children: (
        <Text>My children</Text>
      ),
    });

    const collapse = result.getByTestId('collapse').props.style;

    await waitFor(() => {
      expect(collapse.height).toBe(
        fakeHeaderLayoutEvent.nativeEvent.layout.height + DefaultTheme.spacings.sMedium,
      );
    });
  });

  it('expanded height', async () => {
    const result = await renderCollapse({
      testID: 'collapse',
      title: 'My title',
      description: 'My description',
      children: (
        <Text>My children</Text>
      ),
      expanded: true,
    });

    const collapse = result.getByTestId('collapse').props.style;

    await waitFor(() => {
      expect(collapse.height).toBe(
        fakeContentLayoutEvent.nativeEvent.layout.height,
      );
    });
  });

  it('Collapse animation', async () => {
    const onPressMock = jest.fn();
    const onCollapseEndMock = jest.fn();

    const result = await renderCollapse({
      testID: 'collapse-component',
      onPress: onPressMock,
      title: 'Title Mock',
      description: 'Description Mock',
      onCollapseEnd: onCollapseEndMock,
      children: (
        <Text>Text content</Text>
      ),
    });

    const collapseTouchable = result.getByTestId('collapse-touchable');
    fireEvent.press(collapseTouchable);

    expect(onPressMock).toBeCalled();

    await waitFor(() => {
      expect(onCollapseEndMock).toBeCalled();
    });
  });

  it('Should verify if property numberOfLine is equals to 1', async () => {
    const onPressMock = jest.fn();
    const onCollapseEndMock = jest.fn();

    const result = await renderCollapse({
      testID: 'collapse-component',
      onPress: onPressMock,
      title: 'Title Mock',
      description: 'Description Mock',
      onCollapseEnd: onCollapseEndMock,
      children: (
        <Text>Text content</Text>
      ),
    });

    const collapseTitle = result.getByTestId('collapse-title');

    expect(collapseTitle.props).toHaveProperty('numberOfLines', 1);
  });

  it('Should verify if property numberOfLine is equals to 0', async () => {
    const onPressMock = jest.fn();
    const onCollapseEndMock = jest.fn();

    const result = await renderCollapse({
      testID: 'collapse-component',
      onPress: onPressMock,
      title: 'Title Mock',
      description: 'Description Mock',
      onCollapseEnd: onCollapseEndMock,
      titleNumberOfLines: 0,
      children: (
        <Text>Text content</Text>
      ),
    });

    const collapseTitle = result.getByTestId('collapse-title');

    expect(collapseTitle.props).toHaveProperty('numberOfLines', 0);
  });
});
