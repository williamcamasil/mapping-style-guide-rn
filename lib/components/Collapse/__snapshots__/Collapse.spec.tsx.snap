// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse snapshots children 1`] = `
<View
  collapsable={false}
  style={
    {
      "borderColor": "#F0F0F2",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 74,
      "overflow": "hidden",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="collapse-content"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 24,
          "paddingBottom": 0,
        }
      }
      testID="collapse-touchable"
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
        testID="collapse-title"
      >
        My title
      </Text>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={16}
        vbWidth={16}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m4 6 4 4 4-4"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      style={
        {
          "marginTop": 8,
          "opacity": 0,
          "paddingHorizontal": 24,
        }
      }
      testID="collapse-description-container"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
        testID="collapse-description"
      >
        My description
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "opacity": 0,
          "padding": 24,
          "paddingTop": 0,
        }
      }
      testID="collapse-children-container"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        My children
      </Text>
    </View>
  </View>
</View>
`;

exports[`Collapse snapshots default 1`] = `
<View
  collapsable={false}
  style={
    {
      "borderColor": "#F0F0F2",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 74,
      "overflow": "hidden",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="collapse-content"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 24,
          "paddingBottom": 0,
        }
      }
      testID="collapse-touchable"
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
        testID="collapse-title"
      >
        My title
      </Text>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={16}
        vbWidth={16}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m4 6 4 4 4-4"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      style={
        {
          "marginTop": 8,
          "opacity": 0,
          "paddingHorizontal": 24,
        }
      }
      testID="collapse-description-container"
    />
    <View
      collapsable={false}
      style={
        {
          "opacity": 0,
          "padding": 24,
          "paddingTop": 0,
        }
      }
      testID="collapse-children-container"
    />
  </View>
</View>
`;

exports[`Collapse snapshots description 1`] = `
<View
  collapsable={false}
  style={
    {
      "borderColor": "#F0F0F2",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 74,
      "overflow": "hidden",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="collapse-content"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 24,
          "paddingBottom": 0,
        }
      }
      testID="collapse-touchable"
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
        testID="collapse-title"
      >
        My title
      </Text>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={16}
        vbWidth={16}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="m4 6 4 4 4-4"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      style={
        {
          "marginTop": 8,
          "opacity": 0,
          "paddingHorizontal": 24,
        }
      }
      testID="collapse-description-container"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
        testID="collapse-description"
      >
        My description
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "opacity": 0,
          "padding": 24,
          "paddingTop": 0,
        }
      }
      testID="collapse-children-container"
    />
  </View>
</View>
`;

exports[`Collapse snapshots expanded 1`] = `
<View
  collapsable={false}
  style={
    {
      "borderColor": "#F0F0F2",
      "borderRadius": 12,
      "borderWidth": 1,
      "height": 200,
      "overflow": "hidden",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="collapse-content"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 24,
          "paddingBottom": 0,
        }
      }
      testID="collapse-touchable"
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
            "marginRight": 4,
          }
        }
        testID="collapse-title"
      >
        My title
      </Text>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={16}
        vbWidth={16}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M12 10 8 6l-4 4"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4290954697,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      collapsable={false}
      style={
        {
          "marginTop": 8,
          "opacity": 1,
          "paddingHorizontal": 24,
        }
      }
      testID="collapse-description-container"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
        testID="collapse-description"
      >
        My description
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "opacity": 1,
          "padding": 24,
          "paddingTop": 0,
        }
      }
      testID="collapse-children-container"
    >
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#676D79",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "normal",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        My children
      </Text>
    </View>
  </View>
</View>
`;
